core/src/main/java/io/grpc/internal/ServiceConfigUtil.java

```diff
@@ -345,52 +367,86 @@ static Integer getMaxResponseMessageBytesFromMethodConfig(Map<String, Object> me
   }
 
   /**
-   * Extracts the loadbalancing policy name from loadbalancer config.
+   * Unwrap a LoadBalancingConfig JSON object into a {@link LbConfig}.  The input is a JSON object
+   * (map) with exactly one entry, where the key is the policy name and the value is a config object
+   * for that policy.
    */
-  public static String getBalancerPolicyNameFromLoadBalancingConfig(Map<String, Object> lbConfig) {
-    return lbConfig.entrySet().iterator().next().getKey();
+  @SuppressWarnings("unchecked")
+  public static LbConfig unwrapLoadBalancingConfig(Object lbConfig)
+      throws MalformedConfigException {
+    Map<String, Object> map;
+    try {
+      map = (Map<String, Object>) lbConfig;
+    } catch (ClassCastException e) {
+      throw new MalformedConfigException("Invalid type. Config=" + lbConfig, e);
+    }
+    if (map.size() != 1) {
+      throw new MalformedConfigException(
+          "There are " + map.size() + " fields in a LoadBalancingConfig object. Exactly one"
+          + " is expected. Config=" + lbConfig);
+    }
+    Map.Entry<String, Object> entry = map.entrySet().iterator().next();
+    Map<String, Object> configValue;
+    try {
+      configValue = (Map<String, Object>) entry.getValue();
+    } catch (ClassCastException e) {
+      throw new MalformedConfigException("Invalid value type.  value=" + entry.getValue(), e);
+    }
+    return new LbConfig(entry.getKey(), configValue);
+  }
+
+  /**
+   * Given a JSON list of LoadBalancingConfigs, and convert it into a list of LbConfig.
+   */
+  @SuppressWarnings("rawtypes")
+  public static List<LbConfig> unwrapLoadBalancingConfigList(Object listObject)
+      throws MalformedConfigException {
+    List list;
```

Comment:
Use `@SuppressWarnings("unchecked")` and `List<Object>` or `List<?>` instead of `@SuppressWarnings("rawtypes")` and `List`?

Reply:
Done.